src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceSetDeserializer.cs

```diff
@@ -115,19 +117,139 @@ public sealed override object ReadInline(object item, IEdmTypeReference edmType,
         /// <param name="readContext">The deserializer context.</param>
         /// <param name="elementType">The element type of the resource set being read.</param>
         /// <returns>The deserialized resource set object.</returns>
-        public virtual IEnumerable ReadResourceSet(ODataResourceSetWrapper resourceSet, IEdmStructuredTypeReference elementType, ODataDeserializerContext readContext)
+        public virtual IEnumerable ReadResourceSet(ODataResourceSetWrapperBase resourceSet, IEdmStructuredTypeReference elementType, ODataDeserializerContext readContext)
         {
             ODataEdmTypeDeserializer deserializer = DeserializerProvider.GetEdmTypeDeserializer(elementType);
             if (deserializer == null)
             {
                 throw new SerializationException(
                     Error.Format(SRResources.TypeCannotBeDeserialized, elementType.FullName()));
             }
+                       
+            if (resourceSet.ResourceSetType == ResourceSetType.ResourceSet)
+            {
+                foreach (ODataResourceWrapper resourceWrapper in resourceSet.Resources)
+                {
+                    yield return deserializer.ReadInline(resourceWrapper, elementType, readContext);
+                }
+            }
+            else
+            {
+                foreach (ODataItemBase odataItemBase in resourceSet.Resources)
+                {
+                    ODataResourceWrapper resourceWrapper = odataItemBase as ODataResourceWrapper;
+
+                   if(resourceWrapper != null)
+                    {
+                        ODataDeletedResource deletedResource = resourceWrapper.ResourceBase as ODataDeletedResource;
+
+                        if (deletedResource != null)
+                        {
+                            yield return DeSerializeDeletedEntity(readContext, deletedResource);
```

Comment:
>readContext, deletedResource [](start = 66, length = 28)

probably more consistent with ReadInline to put the resource wrapper (i.e., deletedResource) parameter first, and the readContext last.